compiler/src/dotty/tools/dotc/typer/Implicits.scala

```diff
@@ -702,11 +702,6 @@ trait Implicits { self: Typer =>
       }
     }
 
-  lazy val synthesizedTastyContext: SpecialHandler =
-    (formal, span) => implicit ctx =>
-      if (ctx.inInlineMethod || enclosingInlineds.nonEmpty) ref(defn.TastyReflection_macroContext)
-      else EmptyTree
-
```

Comment:
Is it too early to remove it? Or we may assume that if people are experimenting with new macros, they can tolerate such changes without a transition period?

Reply:
For most things we already require a `QuoteContext` with a stable path which this implicit conversion does not provide. Hence using `Reflection` may easily lead to confusing errors. Having a single way to do things will be clearer. This change also aims to align fully with the documentation and fail where the code is using a pattern that will probably not work at long terms.